### PR TITLE
ISSUE #3727 - V5 Teamspace user list MVP

### LINKED ISSUE #3727: V5 Teamspace user list MVP
### Description

In order to get a functional version of V5 up and running we must have a means to access the teamspace user list, including adding and removing users

### Goals

- [ ] Allow the user list to be accessed from V5
- [ ] Where possible maintain styling with V5
- [ ] If the page opens in a new tab, avoid users from accessing other elements of V4

### PR DESCRIPTION
**Pull request by The-Daniel**
This fixes #3727

#### Description
Adds V5 style to the teamspace user settings
This also fixes ISSUE #3635

#### Test cases

Make sure you check each of the following:
- Add user Popover
- Add user popover when trying to add a user that doesn't exist
- Invite user modal empty
- Invite user modal with project selected
- Invitees list filled
(click here) image
- Invitees list empty
- Delete teamspace admin confirmation modal

### CODE REVIEW

**Comment by george-hadfield** [comment]
Thanks @The-Daniel. I'm going to spend some time going through this is more detail but overall is looking good - good job.

**Comment by george-hadfield** [comment]
### Add "Users" title to top left of white box
To match projects design, please add a "Users" title to the top left of the white box

Build
image

Projects screen
image

**Comment by george-hadfield** [comment]
### Move search and user to left
Left side of search icon should align with left side of users first name

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Add sort arrow to column titles

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Reduce space between column titles and top of list

Ensure it matches the spacing used in the federations and containers screens

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Change circles in job menu to match size of selected job circle

image

**Comment by george-hadfield** [comment]
### Change font colour and size in role and permissions menu's

The menu styles should match that of the other menus such as the one from the Federations ellipsis

Build
image

Menu it should match
image

**Comment by george-hadfield** [comment]
### Increase font size of "Assigned licenses..." to 12px

image

**Comment by george-hadfield** [comment]
### Align left of magnifiying glass inline with titles

image

**Comment by george-hadfield** [comment]
### Move add user button up
Should have the same positioning as the "Add project" button on the projects page. flick between tabs and you'll see the difference


image

**Comment by george-hadfield** [comment]
### Teamspace admin single select menu needs styling the same as all other menus

image

**Comment by george-hadfield** [comment]
### The internal borders are a different colour to the outside borders

image

Should match the border styling of federations and containers

**Comment by george-hadfield** [comment]
### Reduce spacing at the bottom of white container
Should be about half of what is there currently

image

**Comment by george-hadfield** [comment]
### Invite user modal should cancel the add user modal

image

**Comment by george-hadfield** [comment]
### Fix spacing in modal save bar at the bottom
Should match other modals, as well as the styling

image

**Comment by george-hadfield** [comment]
### Centre align column titles

image

**Comment by george-hadfield** [comment]
### Change "Add project permissions" to button

Design
image

Build
image

**Comment by george-hadfield** [comment]
### Cancel button bigger than Remove button on delete user modal

image

**Review by george-hadfield — CHANGES_REQUESTED** [review]
Comments left with a few changes

**Comment by george-hadfield** [comment]
### Should remove user use the same modal as remove project?

image

**Comment by The-Daniel** [comment]
> ### Change "Add project permissions" to button
> Design image
> 
> Build image

This is the same button seen in[ the modal before selecting a project.](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1663%3A13734&t=M6YTfSHnRsmtNeyC-1) Which design do you want?

**Comment by george-hadfield** [comment]
> > ### Change "Add project permissions" to button
> > Design image
> > Build image
> 
> This is the same button seen in[ the modal before selecting a project.](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=1663%3A13734&t=M6YTfSHnRsmtNeyC-1) Which design do you want?

Can we go with button please. My bad.

**Comment by sanmont3drepo** [comment]
- When searching the USER label is still there 

image

**Comment by carmenfan** [comment]
> > > The link in the invitation has the v4 registration. Should we change this in this PR @carmenfan ?
> > 
> > 
> > You mean the registration invite email?
> 
> Yup, when you gate the invitarion the link redirect to https://3drepo.local/sign-up?email=anemail@mailinator.com

@mosment3DRepo  how much do you care about this? Do we want them just be redirected to v5 regardless, or v4 if it came from v4, and v5 if it came from v5?

File: frontend/src/v5/ui/themes/theme.ts [hunk 1]
```diff
@@ -432,7 +432,13 @@ export const theme = createTheme({
 					color: null,
 				},
 				colorPrimary: {
-					color: COLOR.PRIMARY_MAIN,
+					color: COLOR.BASE_LIGHT,
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/themes/theme.ts:435]: What is this for? (just checking this doesn't impact other components )

File: frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import { Cell, Head, Row } from '@/v4/routes/components/customTable/customTable.styles';
+import { FloatingButton, FloatingButtonContainer } from '@/v4/routes/components/floatingActionPanel/floatingActionPanel.styles';
+import { Content as LoadingText } from '@/v4/routes/components/loader/loader.styles';
+import { Container as UserTable, Footer as LicencesFooter } from '@/v4/routes/components/userManagementTab/userManagementTab.styles';
+import { LoaderContainer } from '@/v4/routes/userManagement/userManagement.styles';
+import { PendingInvites } from '@/v4/routes/users/users.styles';
+import styled, { css } from 'styled-components';
+
+const AddUserButton = css`
+	${FloatingButtonContainer} {
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts:27]: Shouldn't this overrides for v4 be in overrides?

File: frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx [hunk 3]
```diff
@@ -37,9 +37,19 @@ interface IDeleteModal {
...
 
-export const DeleteModal = ({ onClickConfirm, onClickClose, name, message, confidenceCheck }: IDeleteModal) => {
+export const DeleteModal = ({
+	onClickConfirm,
+	onClickClose,
+	name,
+	message,
+	confidenceCheck,
+	titleLabel,
+	confirmLabel,
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx:51]: I dont get this in the component is being used as boolean only.

File: frontend/src/v4/routes/components/highlight/highlight.styles.ts [hunk 4]
```diff
@@ -15,11 +15,12 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+import { isV5 } from '@/v4/helpers/isV5';
 import styled from 'styled-components';
 import { FONT_WEIGHT } from '../../../styles';
 
 export const Mark = styled.mark`
-	color: #06563c;
+	color: ${isV5() ? ({ theme }) => theme.palette.primary.main : '#06563c'};
```
Comment (sanmont3drepo) [inline frontend/src/v4/routes/components/highlight/highlight.styles.ts:23]: Is too complicated to have this in v4 overrides?

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx (hunk 3 above)
Comment (The-Daniel) [reply frontend/src/v5/ui/components/shared/modals/templates/deleteModal/deleteModal.component.tsx:51]: confirmLabel and titleLabel aren't being used as boolean. If they are provided then they are used. If they are not provided then there is a default that takes over

File: frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts (hunk 2 above)
Comment (The-Daniel) [reply frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts:27]: I was copying what was done in `routes\dashboard\projects\userPermissions\userPermissions.styles.ts`.
I can instead put it in an overrides file if you think that's better

File: frontend/src/v4/routes/components/highlight/highlight.styles.ts (hunk 4 above)
Comment (The-Daniel) [reply frontend/src/v4/routes/components/highlight/highlight.styles.ts:23]: I tried this. The problem is that this component is being rendered outside of the V4Overrides DOM (This is for the highlight that you see when typing a users name into the add user popover )
image
So this seemed like the easiest solution

File: frontend/src/v5/ui/themes/theme.ts (hunk 1 above)
Comment (The-Daniel) [reply frontend/src/v5/ui/themes/theme.ts:435]: image
It's for these radio controls. (Can also be seen in Project Permissions)

**Comment by george-hadfield** [comment]
### Button width needs fixing

image

**Comment by george-hadfield** [comment]
### Cancel in 'invite user' modal should be an outlined button using primary.main

image

**Comment by george-hadfield** [comment]
### Background colour of bottom bar in modal should be light blue not white

image

**Comment by george-hadfield** [comment]
### Not enough left and right padding in modal

image

**Comment by george-hadfield** [comment]
### Above issues also apply to the "pending licenses" modal

image

File: frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts (hunk 2 above)
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/dashboard/teamspaces/users/usersList.styles.ts:27]: I think that one was before the overrides thing was in created. So yes, if we can lets try to put in there. My guess is that if you place it in overrides, the one from userpermissions is no longer necessary?

**Comment by george-hadfield** [comment]
### User and job should be in line horizontally

**Review by george-hadfield — DISMISSED** [review]

**Review by sanmont3drepo — DISMISSED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**